### ISSUE TITLE
RecurrenceField defaults to empty Recurrence object rather than none

### ISSUE THREAD

**Issue by GeeWee** [django-recurrence/django-recurrence#136]
Test case:

```python
class RecurringOrderGenerator(models.Model):
    recurrences = RecurrenceField(blank=False, null=False)
```
```python
# Test
order = RecurringOrderGenerator()
order.full_clean()
# Expecting this to fail with an error that occurrences must be present
# Instead passes. RecurringOrderGenerator now has an empty recurrence object.
```

I've looked at the code and I think the issue is in 
`recurrence/base.py : deserialize`
It will return an empty occurrence on an empty string, but it should return `None`.
```python
if text == '':
     Return None
```

I wouldn't mind submitting a PR, but note that this is a breaking change of the public API - some people might be depending on this bug.

**Comment by dominicrodger** [comment]
PR welcome - I think now we're more definite about allowing fields to be blank, or requiring them not to be (#133), this change makes sense. We can mark it as a breaking change with a version bump.

**Comment by GeeWee** [comment]
Alright, I've been working a bit on this, and I think we need to specify the semantics of the field a little bit for the fix to get good.
I'm looking for input on how to handle blanks and null in the fields. Seeing as we extend 

Currently we save empty periods as empty strings '' in the database, but I assume we want these to be "None" when they're retrieved again.

I'm thinking the following behaviour is appropriate:
## Into DB
 - We save `''` as `''` in the database.
 - We save `None` as `Null` in the database. User has to deal with the `null` constraint themselves
 - We save Recurrences in string format as regular

## From DB
- We convert `''` to `None` when taking it out of the database
- Other strings are converted to Recurrence events when taken out of the database
- None is returned as None

Thoughts?

**Comment by dominicrodger** [comment]
Hey @GeeWee - appreciate all the thought you're putting in to this.

I think we should save empty periods as null values - although we'll have to convert `''` into `None` for data that's already stored. Can anyone think of a downside to this? Any thoughts here @AimeeMS?

**Comment by GeeWee** [comment]
I agree that that makes the most sense. To make sure I understand, you want to do the `''` -> `None` conversion only from the database and out, to make sure that empty periods that are already saved, are handled correctly?

**Comment by dominicrodger** [comment]
Hey @GeeWee - if an instance has an empty value, we should save it to the database as null. When we retrieve null or empty values from the database, we should convert to `None`. Hopefully with this change, the only time we'd see empty values would be from old versions of django-recurrence.

Does that make sense?

**Comment by GeeWee** [comment]
Yeah that makes sense to me.
